src/main/java/com/fleotadezuta/youthprogrammanager/facade/ChildParentFacade.java

```diff
@@ -0,0 +1,211 @@
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@AllArgsConstructor
+public class ChildParentFacade {
+    private final ChildRepository childRepository;
+    private final ChildMapper childMapper;
+    private final ParentMapper parentMapper;
+    private final ParentService parentService;
+    private final ChildService childService;
+
+    public Flux<ParentDto> getPotentialParents(String name) {
+        return parentService.findByFullName(name);
+    }
+
+    public Flux<ChildDto> getPotentialChildren(String name) {
+        return childService.findByFullName(name);
+    }
+
+    public Flux<ParentUpdateDto> getAllParents() {
+        return parentService.findAll()
+                .flatMap(parent ->
+                        childService.findByParentId(parent.getId())
+                                .map(ChildDocument::getId)
+                                .collectList()
+                                .map(childIds -> {
+                                    ParentUpdateDto parentUpdateDto = parentMapper.fromParentDocumentToParentUpdateDto(parent);
+                                    parentUpdateDto.setChildIds(childIds);
+                                    return parentUpdateDto;
+                                })
+                );
+    }
+
+
+    public Mono<ChildDto> addChild(ChildDto childDto) {
+        List<RelativeParents> relativeParents = childDto.getRelativeParents();
+        if (relativeParents == null || relativeParents.isEmpty()) {
+            ChildDocument childDocument = childMapper.fromChildDtoToChildDocument(childDto);
+            return childRepository.save(childDocument)
+                    .map(childMapper::fromChildDocumentToChildDto);
+        }
+        List<Mono<ParentDocument>> parentMonos = relativeParents.stream()
+                .map(parent -> parentService.findById(parent.getId())
+                        .switchIfEmpty(Mono.error(new IllegalArgumentException("Invalid parentId: " + parent.getId()))))
+                .collect(Collectors.toList());
+
+        return Flux.merge(parentMonos)
+                .collectList()
+                .flatMap(parents -> {
+                    ChildDocument childDocument = childMapper.fromChildDtoToChildDocument(childDto);
+                    return childRepository.save(childDocument)
+                            .map(childMapper::fromChildDocumentToChildDto);
+                })
+                .onErrorResume(Mono::error);
+    }
+
+    public Mono<ParentDto> deleteParent(String id) {
+        return parentService.findById(id)
+                .flatMap(parent -> parentService.deleteById(id)
+                        .then(updateChildren(parent.getId()))
+                        .thenReturn(parent))
+                .map(parentMapper::fromParentDocumentToParentDto);
+    }
+
+    private Mono<Void> updateChildren(String parentIdToRemove) {
```

Comment:
Ez inkabb `removeParentFromChildren` vagy valami ilyesmi. (es akar ChildServiceben is lehet)